perfmetrics/scripts/ls_metrics/listing_benchmark.py

```diff
@@ -0,0 +1,452 @@
+  subprocess.call('mkdir {}'.format(gcs_bucket), shell=True)
+
+  exit_code = subprocess.call('gcsfuse --implicit-dirs --disable-http2 --max-conns-per-host 100 {} {}'.format(
+      bucket_name, gcs_bucket), shell=True)
+  if exit_code != 0:
+    log.error('Cannot mount the GCS bucket due to exit code %s.\n', exit_code)
+    subprocess.call('bash', shell=True)
+  else:
+    return gcs_bucket
+
+
+def _parse_arguments(argv):
+  """Parses the arguments provided to the script via command line.
+
+  Args:
+    argv: List of arguments recevied by the script.
+
+  Returns:
+    A class containing the parsed arguments.
+  """
+
+  argv = sys.argv
+  parser = argparse.ArgumentParser()
+  parser.add_argument(
+      'config_file',
+      help='Provide path of the config file.',
+      action='store'
+  )
+  parser.add_argument(
+      '--keep_files',
+      help='Does not delete the directory structure in persistent disk.',
+      action='store_true',
+      default=False,
+      required=False,
+  )
+  parser.add_argument(
+      '--upload',
+      help='Upload the results to the Google Sheet.',
+      action='store_true',
+      default=False,
+      required=False,
+  )
+  parser.add_argument(
+      '--message',
+      help='Puts a message/title describing the test.',
+      action='store',
+      nargs=1,
+      default='Performance Listing Benchmark',
+      required=False,
+  )
+  parser.add_argument(
+      '--num_samples',
+      help='Number of samples to collect of each test.',
+      action='store',
+      nargs=1,
+      default=[10],
+      required=False,
+  )
+  parser.add_argument(
+      '--command',
+      help='Command to run the tests on.',
+      action='store',
+      nargs=1,
+      default=['ls -R'],
+      required=True,
+  )
+  # Ignoring the first parameter, as it is the path of this python script itself.
+  return parser.parse_args(argv[1:])
+
+
+def _check_dependencies(packages) -> None:
+  """Check whether the dependencies are installed or not.
+
+  Args:
+    packages: List containing the names of the dependencies to be checked.
+
+  Raises:
+    Aborts the execution if a particular dependency is not found.
+  """
+
+  for curr_package in packages:
+    log.info('Checking whether %s is installed.\n', curr_package)
+    exit_code = subprocess.call(
+        '{} --version'.format(curr_package), shell=True)
+    if exit_code != 0:
+      log.error('%s not installed. Please install. Aborting!\n', curr_package)
+      subprocess.call('bash', shell=True)
+
+  return
+
+
+if __name__ == '__main__':
+  argv = sys.argv
+  if len(argv) < 3:
+    raise TypeError('Incorrect number of arguments.\n'
+                    'Usage: '
+                    'python3 listing_benchmark.py [--keep_files] [--upload] [--num_samples NUM_SAMPLES] [--message MESSAGE] --command COMMAND config_file')
+
+  args = _parse_arguments(argv)
+
+  _check_dependencies(['gsutil', 'gcsfuse'])
+
+  f = open(os.path.abspath(args.config_file))
```

Comment:
you can use with aswell. explicit closing of file handler is not required.

Reply:
done